cpp/proto/cert_serializer.cc

```diff
@@ -438,17 +451,17 @@ SerializeResult SerializeV2PrecertSCTMerkleTreeLeaf(
   if (res != SerializeResult::OK) {
     return res;
   }
-  TLSSerializer serializer;
-  serializer.WriteUint(ct::V2, Serializer::kVersionLengthInBytes);
-  serializer.WriteUint(ct::TIMESTAMPED_ENTRY,
-                       Serializer::kMerkleLeafTypeLengthInBytes);
-  serializer.WriteUint(timestamp, Serializer::kTimestampLengthInBytes);
-  serializer.WriteUint(ct::PRECERT_ENTRY_V2,
-                       Serializer::kLogEntryTypeLengthInBytes);
-  serializer.WriteFixedBytes(issuer_key_hash);
-  serializer.WriteVarBytes(tbs_certificate, kMaxCertificateLength);
-  serializer.WriteSctExtension(sct_extension);
-  result->assign(serializer.SerializedString());
+  result->clear();
+  WriteUint(ct::V2, Serializer::kVersionLengthInBytes, result);
+  WriteUint(ct::TIMESTAMPED_ENTRY,
+                           Serializer::kMerkleLeafTypeLengthInBytes, result);
+  WriteUint(timestamp, Serializer::kTimestampLengthInBytes,
+                           result);
+  WriteUint(ct::PRECERT_ENTRY_V2,
+                           Serializer::kLogEntryTypeLengthInBytes, result);
```

Comment:
`clang-format` might need to be "helped" a bit for the previous few lines (put everything on single lines, and re-run it).